src/mem/alloc.h

```diff
@@ -214,35 +214,6 @@ namespace snmalloc
 #endif
     }
 
-    /**
-     * Checks the allocation at `p` could have been validly allocated with
-     * a size of `size`.
-     */
-    void check_size(void* p, size_t size)
-    {
-#if defined(CHECK_CLIENT)
-      auto asize = alloc_size(p);
-      auto asc = size_to_sizeclass(asize);
-      if (size_to_sizeclass(size) != asc)
```

Comment:
This functionality appears to be lost with this change.  You are no longer checking that deallocations are the correct size class just whether they are the correct size kind (large, ,medium or small).

Reply:
Also, I now wonder if this code works for large classes.  In particular, does 
```
free(malloc(SUPERSLAB_SIZE+1))
```
work with `CHECK_CLIENT`.

Reply:
I surely wasn't intending to remove this check!  The `{small,medium}_dealloc_checked_chunkmap` and `large_dealloc_unchecked` functions are comparing the sizeclass derived from the provided size against the sizeclass in the chunkmap (for large allocs) or slab headers (for the rest).  What'd I miss?

Reply:
You're right.  I do wonder about adding tests for this.  Having a Custom Pal that intercepts the Error just, and then calling with a bunch of correct and incorrect deallocations.  Can be part of a separate PR. I am happy with this one.